app/models/npq_applications/eligibility_import.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module NPQApplications
+  class EligibilityImport < ApplicationRecord
+    self.table_name = "npq_application_eligibility_imports"
+
+    belongs_to :user, inverse_of: :npq_application_eligibility_imports
+
+    validates :filename, presence: { message: "Filename can't be blank" }
+
+    validates :status, presence: true
+
+    validates :user, presence: true
+    validate :validate_user_is_admin
+
+    enum status: {
+      pending: "pending",
+      processing: "processing",
+      completed: "completed",
+      completed_with_errors: "completed_with_errors",
+      failed: "failed",
+    }
+
+    scope :new_to_old, -> { order(created_at: :desc) }
+
+    def perform_later
+      Admin::NPQApplications::EligibilityImportJob.perform_later(self)
+    end
+
+    def done?
+      completed? || failed? || completed_with_errors?
+    end
```

Comment:
The word 'done' suggests success to me, could `#finished?` work here?

Reply:
I went with processed to mirror the processed_at attribute.